mm: bot panics (trying to placeMultiTrade)

MM bot panic (note, I'm running my own fork at https://github.com/norwnd/dcrdex/commits/master - but this is probably relevant for thus **upstream** repo too):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x102a4c268]

goroutine 4980 [running]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).placeMultiTrade(0x14006ba6600, {0x1400b60a0a0, 0x1, 0x10383541c?}, 0x1)
	/Users/norwnd/dcrdex/client/mm/exchange_adaptor.go:970 +0x438
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).multiTrade(0x14006ba6600, {0x1400b60a088, 0x1, 0x1400b608150?}, 0x1, 0x0, 0x6e4f8d3)
	/Users/norwnd/dcrdex/client/mm/exchange_adaptor.go:1343 +0xc60
decred.org/dcrdex/client/mm.(*basicMarketMaker).rebalance(0x140087e20c0, 0x6e4f8d3)
	/Users/norwnd/dcrdex/client/mm/mm_basic.go:590 +0x274
decred.org/dcrdex/client/mm.(*basicMarketMaker).botLoop.func1()
	/Users/norwnd/dcrdex/client/mm/mm_basic.go:634 +0x294
created by decred.org/dcrdex/client/mm.(*basicMarketMaker).botLoop in goroutine 3828
	/Users/norwnd/dcrdex/client/mm/mm_basic.go:619 +0x26c
make: *** [a] Error 2
```

I think `MultiTrade` func needs to be adjusted like this:
```go
// MultiTrade is used to place multiple standing limit orders on the same
// side of the same market simultaneously.
func (c *Core) MultiTrade(pw []byte, form *MultiTradeForm) []*MultiTradeResult {
	results := make([]*MultiTradeResult, 0, len(form.Placements))
	reqs, err := c.prepareMultiTradeRequests(pw, form)
	if err != nil {
		for range form.Placements {
			results = append(results, &MultiTradeResult{Error: err})
		}
		return results
	}

	for _, req := range reqs {
		var corder *Order
		corder, err = c.sendTradeRequest(req)
		if err != nil {
			results = append(results, &MultiTradeResult{Error: err})
			continue
		}
		results = append(results, &MultiTradeResult{Order: corder})
	}

	return results
}
```
there is a related "sanity-check" in MM code after `MultiTrade` is called (but it's always gonna pass with `master` code version):
```go
	if len(placements) != len(results) {
		u.log.Errorf("unexpected number of results. expected %d, got %d", len(placements), len(results))
		return results
	}
```
not sure if all that ^ means there is a deeper underlying issue, but `panic` is certainly not a desired outcome.

I'm getting what seems like the same error on a vanilla decred v2.0.5 install:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xaae6ed]

goroutine 11569 [running]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).placeMultiTrade(0xc00270b000, {0xc01f18aae0, 0x3, 0xc000dd9860?}, 0x1)
        decred.org/dcrdex@v1.0.2/client/mm/exchange_adaptor.go:970 +0x52d
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).multiTrade(0xc00270b000, {0xc00ed492d8, 0x3, 0xc001012bc0?}, 0x1, 0x3f60624dd2f1a9fc, 0x6e59a8d)
        decred.org/dcrdex@v1.0.2/client/mm/exchange_adaptor.go:1342 +0x104c
decred.org/dcrdex/client/mm.(*basicMarketMaker).rebalance(0xc000c1df88?, 0x6e59a8d)
        decred.org/dcrdex@v1.0.2/client/mm/mm_basic.go:392 +0x26d
decred.org/dcrdex/client/mm.(*basicMarketMaker).botLoop.func1()
        decred.org/dcrdex@v1.0.2/client/mm/mm_basic.go:429 +0x185
created by decred.org/dcrdex/client/mm.(*basicMarketMaker).botLoop in goroutine 11466
        decred.org/dcrdex@v1.0.2/client/mm/mm_basic.go:421 +0x275
```

My DCRDEX score keeps suffering because of this...

--- a/client/mm/config.go
+++ b/client/mm/config.go
@@ -3,6 +3,7 @@ package mm
 import (
 	"encoding/json"
 	"fmt"
+	"strconv"
 )
 
 // MarketMakingConfig is the overall configuration of the market maker.
@@ -125,6 +126,23 @@ func (c *BotConfig) requiresPriceOracle() bool {
 
 func (c *BotConfig) requiresCEX() bool {
 	return c.SimpleArbConfig != nil || c.ArbMarketMakerConfig != nil
+}
+
+// multiSplitBuffer returns the additional buffer to add to the order size
+// when doing a multi-split. This only applies to the quote asset.
+func (c *BotConfig) multiSplitBuffer() float64 {
+	if c.QuoteWalletOptions == nil {
+		return 0
+	}
+	multiSplitBuffer, ok := c.QuoteWalletOptions["multisplitbuffer"]
+	if !ok {
+		return 0
+	}
+	multiSplitBufferFloat, err := strconv.ParseFloat(multiSplitBuffer, 64)
+	if err != nil {
+		return 0
+	}
+	return multiSplitBufferFloat
 }
 
 // maxPlacements returns the max amount of placements this bot will place on

--- a/client/mm/exchange_adaptor.go
+++ b/client/mm/exchange_adaptor.go
@@ -1229,14 +1229,19 @@ func (u *unifiedExchangeAdaptor) multiTrade(
 
 	rateCausesSelfMatch := u.rateCausesSelfMatchFunc(sell)
 
+	multiSplitBuffer := u.botCfg().multiSplitBuffer()
+
 	fundingReq := func(rate, lots, counterTradeRate uint64) (dexReq map[uint32]uint64, cexReq uint64) {
 		qty := u.lotSize * lots
+		swapFees := fees.Swap * lots
 		if !sell {
 			qty = calc.BaseToQuote(rate, qty)
+			qty = uint64(math.Round(float64(qty) * (100 + multiSplitBuffer) / 100))
+			swapFees = uint64(math.Round(float64(swapFees) * (100 + multiSplitBuffer) / 100))
 		}
 		dexReq = make(map[uint32]uint64)
 		dexReq[fromID] += qty
-		dexReq[fromFeeID] += fees.Swap * lots
+		dexReq[fromFeeID] += swapFees
 		if u.isAccountLocker(fromID) {
 			dexReq[fromFeeID] += fees.Refund * lots
 		}

--- a/client/mm/exchange_adaptor_test.go
+++ b/client/mm/exchange_adaptor_test.go
@@ -1106,6 +1106,10 @@ func TestMultiTrade(t *testing.T) {
 
 	b2q := calc.BaseToQuote
 
+	addBuffer := func(qty uint64, buffer float64) uint64 {
+		return uint64(math.Round(float64(qty) * (100 + buffer) / 100))
+	}
+
 	/*
 	 * The dexBalance and cexBalances fields of this test are set so that they
 	 * are at an edge. If any non-zero balance is decreased by 1, the behavior
@@ -1116,6 +1120,8 @@ func TestMultiTrade(t *testing.T) {
 		name    string
 		baseID  uint32
 		quoteID uint32
+
+		multiSplitBuffer float64
 
 		sellDexBalances   map[uint32]uint64
 		sellCexBalances   map[uint32]uint64
@@ -1526,6 +1532,392 @@ func TestMultiTrade(t *testing.T) {
 					0: b2q(buyPlacements[1].Rate, lotSize) +
 						b2q(buyPlacements[2].Rate, 2*lotSize) +
 						3*buyFees.Max.Swap + buyFees.Funding,
+				},
+				UsedCEXBal: 3 * lotSize,
+			},
+			expectedOrderIDsWithDecrement: []order.OrderID{
+				orderIDs[4], orderIDs[5],
+			},
+		},
+		{
+			name:    "non account locker - multi split buffer",
+			baseID:  42,
+			quoteID: 0,
+
+			multiSplitBuffer: 0.1,
+
+			// ---- Sell ----
+			sellDexBalances: map[uint32]uint64{
+				42: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding,
+				0:  0,
+			},
+			sellCexBalances: map[uint32]uint64{
+				42: 0,
+				0: b2q(sellPlacements[0].CounterTradeRate, lotSize) +
+					b2q(sellPlacements[1].CounterTradeRate, 2*lotSize) +
+					b2q(sellPlacements[2].CounterTradeRate, 3*lotSize) +
+					b2q(sellPlacements[3].CounterTradeRate, 2*lotSize),
+			},
+			sellPlacements:    sellPlacements,
+			sellPendingOrders: pendingOrders(true, 42, 0),
+			expectedSellPlacements: []*core.QtyRate{
+				{Qty: lotSize, Rate: sellPlacements[1].Rate},
+				{Qty: 2 * lotSize, Rate: sellPlacements[2].Rate},
+				{Qty: lotSize, Rate: sellPlacements[3].Rate},
+			},
+			expectedSellOrderReport: &OrderReport{
+				Placements: []*TradePlacement{
+					{Lots: 1, Rate: sps[0].Rate, CounterTradeRate: sps[0].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX:  map[uint32]uint64{},
+						UsedDEX:      map[uint32]uint64{},
+						RequiredCEX:  0,
+						UsedCEX:      0,
+					},
+					{Lots: 2, Rate: sps[1].Rate, CounterTradeRate: sps[1].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						UsedDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						RequiredCEX: b2q(sellPlacements[1].CounterTradeRate, lotSize),
+						UsedCEX:     b2q(sellPlacements[1].CounterTradeRate, lotSize),
+						OrderedLots: 1,
+					},
+					{Lots: 3, Rate: sps[2].Rate, CounterTradeRate: sps[2].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: 2 * (lotSize + sellFees.Max.Swap),
+						},
+						UsedDEX: map[uint32]uint64{
+							42: 2 * (lotSize + sellFees.Max.Swap),
+						},
+						RequiredCEX: b2q(sellPlacements[2].CounterTradeRate, 2*lotSize),
+						UsedCEX:     b2q(sellPlacements[2].CounterTradeRate, 2*lotSize),
+						OrderedLots: 2,
+					},
+					{Lots: 2, Rate: sps[3].Rate, CounterTradeRate: sps[3].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						UsedDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						RequiredCEX: b2q(sellPlacements[3].CounterTradeRate, lotSize),
+						UsedCEX:     b2q(sellPlacements[3].CounterTradeRate, lotSize),
+						OrderedLots: 1,
+					},
+				},
+				Fees: sellFees,
+				AvailableDEXBals: map[uint32]*BotBalance{
+					42: {
+						Available: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding,
+					},
+					0: {},
+				},
+				RequiredDEXBals: map[uint32]uint64{
+					42: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding,
+				},
+				RemainingDEXBals: map[uint32]uint64{
+					42: 0,
+					0:  0,
+				},
+				AvailableCEXBal: &BotBalance{
+					Available: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[2].CounterTradeRate, 2*lotSize) +
+						b2q(sellPlacements[3].CounterTradeRate, lotSize),
+					Reserved: b2q(sellPlacements[0].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[2].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[3].CounterTradeRate, lotSize),
+				},
+				RequiredCEXBal: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+					b2q(sellPlacements[2].CounterTradeRate, 2*lotSize) +
+					b2q(sellPlacements[3].CounterTradeRate, lotSize),
+				RemainingCEXBal: 0,
+				UsedDEXBals: map[uint32]uint64{
+					42: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding,
+				},
+				UsedCEXBal: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+					b2q(sellPlacements[2].CounterTradeRate, 2*lotSize) +
+					b2q(sellPlacements[3].CounterTradeRate, lotSize),
+			},
+			expectedSellPlacementsWithDecrement: []*core.QtyRate{
+				{Qty: lotSize, Rate: sellPlacements[1].Rate},
+				{Qty: 2 * lotSize, Rate: sellPlacements[2].Rate},
+			},
+			expectedSellOrderReportWithDEXDecrement: &OrderReport{
+				Placements: []*TradePlacement{
+					{Lots: 1, Rate: sps[0].Rate, CounterTradeRate: sps[0].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX:  map[uint32]uint64{},
+						UsedDEX:      map[uint32]uint64{},
+						RequiredCEX:  0,
+						UsedCEX:      0,
+					},
+					{Lots: 2, Rate: sps[1].Rate, CounterTradeRate: sps[1].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						UsedDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						RequiredCEX: b2q(sellPlacements[1].CounterTradeRate, lotSize),
+						UsedCEX:     b2q(sellPlacements[1].CounterTradeRate, lotSize),
+						OrderedLots: 1,
+					},
+					{Lots: 3, Rate: sps[2].Rate, CounterTradeRate: sps[2].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: 2 * (lotSize + sellFees.Max.Swap),
+						},
+						UsedDEX: map[uint32]uint64{
+							42: 2 * (lotSize + sellFees.Max.Swap),
+						},
+						RequiredCEX: b2q(sellPlacements[2].CounterTradeRate, 2*lotSize),
+						UsedCEX:     b2q(sellPlacements[2].CounterTradeRate, 2*lotSize),
+						OrderedLots: 2,
+					},
+					{Lots: 2, Rate: sps[3].Rate, CounterTradeRate: sps[3].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							42: lotSize + sellFees.Max.Swap,
+						},
+						UsedDEX:     map[uint32]uint64{},
+						RequiredCEX: b2q(sellPlacements[3].CounterTradeRate, lotSize),
+						UsedCEX:     0,
+						OrderedLots: 0,
+					},
+				},
+				Fees: sellFees,
+				AvailableDEXBals: map[uint32]*BotBalance{
+					42: {
+						Available: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding - 1,
+					},
+					0: {},
+				},
+				RequiredDEXBals: map[uint32]uint64{
+					42: 4*lotSize + 4*sellFees.Max.Swap + sellFees.Funding,
+				},
+				RemainingDEXBals: map[uint32]uint64{
+					42: lotSize + sellFees.Max.Swap - 1,
+					0:  0,
+				},
+				AvailableCEXBal: &BotBalance{
+					Available: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[2].CounterTradeRate, 2*lotSize) +
+						b2q(sellPlacements[3].CounterTradeRate, lotSize),
+					Reserved: b2q(sellPlacements[0].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[2].CounterTradeRate, lotSize) +
+						b2q(sellPlacements[3].CounterTradeRate, lotSize),
+				},
+				RequiredCEXBal: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+					b2q(sellPlacements[2].CounterTradeRate, 2*lotSize) +
+					b2q(sellPlacements[3].CounterTradeRate, lotSize),
+				RemainingCEXBal: b2q(sellPlacements[3].CounterTradeRate, lotSize),
+				UsedDEXBals: map[uint32]uint64{
+					42: 3*lotSize + 3*sellFees.Max.Swap + sellFees.Funding,
+				},
+				UsedCEXBal: b2q(sellPlacements[1].CounterTradeRate, lotSize) +
+					b2q(sellPlacements[2].CounterTradeRate, 2*lotSize),
+			},
+
+			// ---- Buy ----
+			buyDexBalances: map[uint32]uint64{
+				42: 0,
+				0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+					b2q(buyPlacements[2].Rate, 2*lotSize)+
+					b2q(buyPlacements[3].Rate, lotSize)+
+					4*buyFees.Max.Swap, 0.1) + buyFees.Funding,
+			},
+			buyCexBalances: map[uint32]uint64{
+				42: 8 * lotSize,
+				0:  0,
+			},
+			buyPlacements:    buyPlacements,
+			buyPendingOrders: pendingOrders(false, 42, 0),
+			expectedBuyPlacements: []*core.QtyRate{
+				{Qty: lotSize, Rate: buyPlacements[1].Rate},
+				{Qty: 2 * lotSize, Rate: buyPlacements[2].Rate},
+				{Qty: lotSize, Rate: buyPlacements[3].Rate},
+			},
+			expectedBuyPlacementsWithDecrement: []*core.QtyRate{
+				{Qty: lotSize, Rate: buyPlacements[1].Rate},
+				{Qty: 2 * lotSize, Rate: buyPlacements[2].Rate},
+			},
+			expectedCancels:              []order.OrderID{orderIDs[2]},
+			expectedCancelsWithDecrement: []order.OrderID{orderIDs[2]},
+			multiTradeResult: []*core.MultiTradeResult{
+				{Order: &core.Order{ID: orderIDs[4][:]}},
+				{Order: &core.Order{ID: orderIDs[5][:]}},
+				{Order: &core.Order{ID: orderIDs[6][:]}},
+			},
+			multiTradeResultWithDecrement: []*core.MultiTradeResult{
+				{Order: &core.Order{ID: orderIDs[4][:]}},
+				{Order: &core.Order{ID: orderIDs[5][:]}},
+			},
+			expectedOrderIDs: []order.OrderID{
+				orderIDs[4], orderIDs[5], orderIDs[6],
+			},
+			expectedBuyOrderReport: &OrderReport{
+				Placements: []*TradePlacement{
+					{Lots: 1, Rate: bps[0].Rate, CounterTradeRate: bps[0].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX:  map[uint32]uint64{},
+						UsedDEX:      map[uint32]uint64{},
+						RequiredCEX:  0,
+						UsedCEX:      0,
+					},
+					{Lots: 2, Rate: bps[1].Rate, CounterTradeRate: bps[1].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						UsedDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						RequiredCEX: lotSize,
+						UsedCEX:     lotSize,
+						OrderedLots: 1,
+					},
+					{Lots: 3, Rate: bps[2].Rate, CounterTradeRate: bps[2].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(2*(b2q(buyPlacements[2].Rate, lotSize)+buyFees.Max.Swap), 0.1),
+						},
+						UsedDEX: map[uint32]uint64{
+							0: addBuffer(2*(b2q(buyPlacements[2].Rate, lotSize)+buyFees.Max.Swap), 0.1),
+						},
+						RequiredCEX: 2 * lotSize,
+						UsedCEX:     2 * lotSize,
+						OrderedLots: 2,
+					},
+					{Lots: 2, Rate: bps[3].Rate, CounterTradeRate: bps[3].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[3].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						UsedDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[3].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						RequiredCEX: lotSize,
+						UsedCEX:     lotSize,
+						OrderedLots: 1,
+					},
+				},
+				Fees: buyFees,
+				AvailableDEXBals: map[uint32]*BotBalance{
+					0: {
+						Available: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+							b2q(buyPlacements[2].Rate, 2*lotSize)+
+							b2q(buyPlacements[3].Rate, lotSize)+
+							4*buyFees.Max.Swap, 0.1) + buyFees.Funding,
+					},
+					42: {},
+				},
+				RequiredDEXBals: map[uint32]uint64{
+					0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+						b2q(buyPlacements[2].Rate, 2*lotSize)+
+						b2q(buyPlacements[3].Rate, lotSize)+
+						4*buyFees.Max.Swap, 0.1) + buyFees.Funding,
+				},
+				RemainingDEXBals: map[uint32]uint64{
+					42: 0,
+					0:  0,
+				},
+				AvailableCEXBal: &BotBalance{
+					Available: 4 * lotSize,
+					Reserved:  4 * lotSize,
+				},
+				RequiredCEXBal:  4 * lotSize,
+				RemainingCEXBal: 0,
+				UsedDEXBals: map[uint32]uint64{
+					0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+						b2q(buyPlacements[2].Rate, 2*lotSize)+
+						b2q(buyPlacements[3].Rate, lotSize)+
+						4*buyFees.Max.Swap, 0.1) + buyFees.Funding,
+				},
+				UsedCEXBal: 4 * lotSize,
+			},
+			expectedBuyOrderReportWithDEXDecrement: &OrderReport{
+				Placements: []*TradePlacement{
+					{Lots: 1, Rate: bps[0].Rate, CounterTradeRate: bps[0].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX:  map[uint32]uint64{},
+						UsedDEX:      map[uint32]uint64{},
+						RequiredCEX:  0,
+						UsedCEX:      0,
+					},
+					{Lots: 2, Rate: bps[1].Rate, CounterTradeRate: bps[1].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						UsedDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						RequiredCEX: lotSize,
+						UsedCEX:     lotSize,
+						OrderedLots: 1,
+					},
+					{Lots: 3, Rate: bps[2].Rate, CounterTradeRate: bps[2].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(2*(b2q(buyPlacements[2].Rate, lotSize)+buyFees.Max.Swap), 0.1),
+						},
+						UsedDEX: map[uint32]uint64{
+							0: addBuffer(2*(b2q(buyPlacements[2].Rate, lotSize)+buyFees.Max.Swap), 0.1),
+						},
+						RequiredCEX: 2 * lotSize,
+						UsedCEX:     2 * lotSize,
+						OrderedLots: 2,
+					},
+					{Lots: 2, Rate: bps[3].Rate, CounterTradeRate: bps[3].CounterTradeRate,
+						StandingLots: 1,
+						RequiredDEX: map[uint32]uint64{
+							0: addBuffer(b2q(buyPlacements[3].Rate, lotSize)+buyFees.Max.Swap, 0.1),
+						},
+						UsedDEX:     map[uint32]uint64{},
+						RequiredCEX: lotSize,
+						UsedCEX:     0,
+						OrderedLots: 0,
+					},
+				},
+				Fees: buyFees,
+				AvailableDEXBals: map[uint32]*BotBalance{
+					0: {
+						Available: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+							b2q(buyPlacements[2].Rate, 2*lotSize)+
+							b2q(buyPlacements[3].Rate, lotSize)+
+							4*buyFees.Max.Swap, 0.1) + buyFees.Funding - 1,
+					},
+					42: {},
+				},
+				RequiredDEXBals: map[uint32]uint64{
+					0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+						b2q(buyPlacements[2].Rate, 2*lotSize)+
+						b2q(buyPlacements[3].Rate, lotSize)+
+						4*buyFees.Max.Swap, 0.1) + buyFees.Funding,
+				},
+				RemainingDEXBals: map[uint32]uint64{
+					42: 0,
+					0:  addBuffer(b2q(buyPlacements[3].Rate, lotSize)+buyFees.Max.Swap, 0.1) - 1,
+				},
+				AvailableCEXBal: &BotBalance{
+					Available: 4 * lotSize,
+					Reserved:  4 * lotSize,
+				},
+				RequiredCEXBal:  4 * lotSize,
+				RemainingCEXBal: lotSize,
+				UsedDEXBals: map[uint32]uint64{
+					0: addBuffer(b2q(buyPlacements[1].Rate, lotSize)+
+						b2q(buyPlacements[2].Rate, 2*lotSize)+
+						3*buyFees.Max.Swap, 0.1) + buyFees.Funding,
 				},
 				UsedCEXBal: 3 * lotSize,
 			},
@@ -2626,6 +3018,12 @@ func TestMultiTrade(t *testing.T) {
 						},
 						eventLogDB: &tEventLogDB{},
 					})
+
+					if test.multiSplitBuffer > 0 {
+						adaptor.botCfg().QuoteWalletOptions = map[string]string{
+							"multisplitbuffer": fmt.Sprintf("%f", test.multiSplitBuffer),
+						}
+					}
 
 					var pendingOrders map[order.OrderID]*pendingDEXOrder
 					if sell {

